### PR TITLE
add "scribble" editor

### LINKED ISSUE #30: stdio [read-only]
Is it possible to make the stdio section editable as altering the output is an old habit of mine.

### PR DESCRIPTION
**Pull request by bqi343**
as in geany

closes #30 

~~also prompts you for a workspace name until you give it one upon creation~~ - only prompts when you creat from dashboard

Is there a way to avoid displaying "Loading..." when switching to/from "scribble"?

### CODE REVIEW

**Comment by bqi343** [comment]
seems that creating a workspace now results in "0 Users Online" D:

**Comment by bqi343** [comment]
Closes #14 as well

**Review by thecodingwizard — COMMENTED** [review]
> Is there a way to avoid displaying "Loading..." when switching to/from "scribble"?

https://github.com/suren-atoyan/monaco-react#props potentially by changing the loading message prop?

**Comment by bqi343** [comment]
In `userSettings.ts`:

```
interface UserSettings {
  name: string;
  color: string;
  editorMode: EditorMode;
  defaultPermission: DefaultPermission;
  defaultLang: Language;
}
```

setter of `userSettingsAtomWithPersistence` syncs with firebase

File: src/atoms/userSettings.ts [hunk 1]
```diff
@@ -1,23 +1,108 @@
...
+  get => get(_userSettingsAtom),
+  (get, _set, val) => {
+    const data: UserSettings = {
+      ...get(_userSettingsAtom),
+      ...val,
+    };
+    const userRef = get(userSettingsRefAtom);
+    if (userRef) userRef.update(data);
+    else {
+      // fail silently, it's probably okay?
```
Comment (bqi343) [inline src/atoms/userSettings.ts:102]: oops this is questionable. Else statement should only be triggered when `val = { defaultLang }` and the URL contains a `queryParam`. In this case, the last viewed lang of the user is not correctly updated.

Specifically, consider the following sequence of steps:

 1. You view a file in Java. 
 2. You open a link that ends with `?lang=py`. 
 3. You create a new workspace in a new tab.

Then the workspace will open to Java, not Python (though Python would make more sense). What's the best way to fix this?

**Comment by bqi343** [comment]
Closes #32

File: src/atoms/userSettings.ts (hunk 1 above)
Comment (thecodingwizard) [reply src/atoms/userSettings.ts:102]: One way is to only allow this to be called after firebase is initialized. I think it's fine as is right now though...

**Pull request merged**